src/Doctrine/ORM/ChannelPricingRepositoryTrait.php

```diff
@@ -14,31 +16,56 @@ trait ChannelPricingRepositoryTrait
 {
     use HasAnyBeenUpdatedSinceTrait;
 
-    public function resetMultiplier(DateTimeInterface $dateTime): void
+    public function resetMultiplier(DateTimeInterface $dateTime): int
     {
         \assert($this instanceof EntityRepository);
 
-        do {
-            $ids = $this->createQueryBuilder('o')
-                ->select('o.id')
-                ->andWhere('o.multiplier != 1')
-                ->setMaxResults(100)
-                ->getQuery()
-                ->getResult()
-            ;
+        $updatedRows = 0;
 
-            $res = (int) $this
-                ->createQueryBuilder('o')
-                ->update()
-                ->set('o.multiplier', 1)
-                ->set('o.updatedAt', ':updatedAt')
-                ->andWhere('o.id IN (:ids)')
-                ->setParameter('updatedAt', $dateTime)
-                ->setParameter('ids', $ids)
-                ->getQuery()
-                ->execute()
-            ;
-        } while ($res > 0);
+        $backOffStrategy = new FibonacciBackOffStrategy(250_000, 5);
+        $tries = 0;
+
+        $connection = $this->_em->getConnection();
+        $oldTransactionIsolation = (int) $connection->getTransactionIsolation();
+        $connection->setTransactionIsolation(TransactionIsolationLevel::READ_COMMITTED);
+
+        do {
+            $connection->beginTransaction();
+
+            try {
+                $ids = $this->createQueryBuilder('o')
+                    ->select('o.id')
+                    ->andWhere('o.multiplier != 1')
+                    ->setMaxResults(100)
+                    ->getQuery()
+                    ->getResult()
+                ;
+
+                $res = (int) $this
+                    ->createQueryBuilder('o')
+                    ->update()
+                    ->set('o.multiplier', 1)
+                    ->set('o.updatedAt', ':updatedAt')
+                    ->andWhere('o.id IN (:ids)')
+                    ->setParameter('updatedAt', $dateTime)
+                    ->setParameter('ids', $ids)
+                    ->getQuery()
+                    ->execute()
+                ;
+
+                $connection->commit();
+
+                $updatedRows += $res;
+            } catch (\Throwable $e) {
+                $connection->rollBack();
+
+                $backOffStrategy->backOff(++$tries, $e);
```

Comment:
Do I even need a backoff strategy since I guess the only possible exception that can be thrown from the above code is a deadlock exception?